File: java/parser/src/main/java/gov/lbl/parser/parser/Element_modificationVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Element_modification;
import gov.lbl.parser.domain.Modification;
import gov.lbl.parser.domain.Name;

public class Element_modificationVisitor extends modelicaBaseVisitor<Element_modification> {
    @Override
    public Element_modification visitElement_modification(modelicaParser.Element_modificationContext ctx) {
        NameVisitor nameVisitor = new NameVisitor();
        Name name = ctx.name() == null ? null : ctx.name().accept(nameVisitor);
        
        ModificationVisitor modificationVisitor = new ModificationVisitor();
        Modification modification = ctx.modification() == null ? null : ctx.modification().accept(modificationVisitor);        

        String_commentVisitor string_commentVisitor = new String_commentVisitor();
        String string_comment = ctx.string_comment() == null ? null : ctx.string_comment().accept(string_commentVisitor);
        
        return new Element_modification(name, modification, string_comment);
    }
}

File: jsParser/parser/Element_modificationVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Element_modification = require('../domain/Element_modification')

const NameVisitor = require('./NameVisitor')
const ModificationVisitor = require('./ModificationVisitor')
const String_commentVisitor = require('./String_commentVisitor')

class Element_modificationVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitElement_modification (ctx) {
    let name = null
    let modification = null
    let string_comment = null

    if (ctx.name()) {
      const nameVisitor = new NameVisitor.NameVisitor()
      name = nameVisitor.visitName(ctx.name())
    }
    if (ctx.modification()) {
      const modificationVisitor = new ModificationVisitor.ModificationVisitor()
      modification = modificationVisitor.visitModification(ctx.modification())
    }
    if (ctx.string_comment()) {
      const string_commentVisitor = new String_commentVisitor.String_commentVisitor()
      string_comment = string_commentVisitor.visitString_comment(ctx.string_comment())
    }

    return new Element_modification.Element_modification(name, modification, string_comment)
  }
};

Element_modificationVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitElement_modification = this.visitElement_modification
exports.Element_modificationVisitor = Element_modificationVisitor
